evennia/commands/cmdset.py

```diff
@@ -559,6 +556,11 @@ def add(self, cmd, allow_duplicates=False):
                 except ValueError:
                     system_commands.append(cmd)
 
+        self.commands = commands
+        if not allow_duplicates:
+            # extra run to make sure to avoid doublets
+            self.commands = list(set(self.commands))
+
```

Comment:
This was originally nested, so it would try to make it unique multiple times, this will only do it once at the end.